### PR TITLE
add new tool tiara

### PR DESCRIPTION
**Pull request by Minamehr**
Hi,
This pull request aims to add a new tool, Tiara. It is a deep-learning-based approach for identifying eukaryotic sequences in metagenomic data.

Kind regards,
Mina

### CODE REVIEW

**Comment by bgruening** [comment]
Tests are unfortunately failng.

File: tools/tiara/macros.xml [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+    <token name="@TOOL_VERSION@">1.0.3</token>
+    <token name="@VERSION_SUFFIX@">galaxy0</token>
+
+    
+
+    <xml name="requirements">
+        <requirements>
+            <requirement type="package" version="@TOOL_VERSION@">tiara</requirement>
+            <requirement type="package" version="3.8">python</requirement>
+            <yield/>
```
Comment (SaimMomin12) [inline tools/tiara/macros.xml:11]: I think this is just being misplaced here?

File: tools/tiara/tiara.xml [hunk 2]
```diff
@@ -0,0 +1,178 @@
...
+        #if $MIN_LEN
+            -m '$MIN_LEN'
+        #end if
+
+        #if $cutoff_stage1
+            -p $cutoff_stage1
+            #if $cutoff_stage2
+            $cutoff_stage2
+            #end if
+        #end if
```
Comment (SaimMomin12) [inline tools/tiara/tiara.xml:36]: Is nested if statement done purposefully here?

File: tools/tiara/tiara.xml [hunk 3]
```diff
@@ -0,0 +1,178 @@
...
+
+        #if $advanced_options.advance.customize_kmer_length == 'customize'
+            --k1 $advanced_options.advance.first_stage_kmer
+            --k2 $advanced_options.advance.second_stage_kmer
+        #end if
+
+    ]]></command>
+    <inputs>
+        <param name="input" type="data" format="fasta" label="input fasta,fasta.gz file"/>
+        <param name="taxonomy_filter" type="select" multiple="true" optional="true" label="Write sequences to fast,fasta.gz files specified in the arguments to this option." help="all refer to all classes present in input fasta (to separate fasta files).">
```
Comment (SaimMomin12) [inline tools/tiara/tiara.xml:46]:
```suggestion
        <param name="taxonomy_filter" type="select" multiple="true" optional="true" label="Write sequences to fasta,fasta.gz files specified in the arguments to this option." help="all refers to all classes present in input fasta (to separate fasta files).">
```

File: tools/tiara/tiara.xml [hunk 4]
```diff
@@ -0,0 +1,178 @@
...
+
+        #if $probabilities
+            --pr '$probabilities'
+        #end if
+        #if $verbose
+            -v '$verbose'
+        #end if
+        #if $MIN_LEN
+            -m '$MIN_LEN'
+        #end if
```
Comment (SaimMomin12) [inline tools/tiara/tiara.xml:29]:
```suggestion
        #if $min_len
            -m '$min_len'
        #end if
```

File: tools/tiara/tiara.xml [hunk 5]
```diff
@@ -0,0 +1,178 @@
...
+            <conditional name="advance">
+                <param name="customize_kmer_length" type="select" label="Advanced options">
+                    <option value="default_options">No, Use param defaults</option>
+                    <option value="customize">Yes, See full parameter list</option>
+                </param>
+                <when value="customize">
+                    <param name="first_stage_kmer" type="select" label="Select k-mer length used in the first stage of classification (Default: 6).">
+                        <option value="4">k-mer length 4</option>
+                        <option value="5">k-mer length 5</option>
+                        <option value="6">default k-mer length</option>
```
Comment (SaimMomin12) [inline tools/tiara/tiara.xml:74]:
```suggestion
                        <option value="6" selected="True">default k-mer length</option>
```

File: tools/tiara/tiara.xml [hunk 6]
```diff
@@ -0,0 +1,178 @@
...
+                    </param>
+                    <param name="second_stage_kmer" type="select" label="k-mer length used in the second stage of classification (Default: 7).">
+                        <option value="4">k-mer length 4</option>
+                        <option value="5">k-mer length 5</option>
+                        <option value="6">k-mer length 6</option>
+                        <option value="7">default k-mer length</option>
```
Comment (SaimMomin12) [inline tools/tiara/tiara.xml:80]:
```suggestion
                        <option value="7" selected="True">default k-mer length</option>
```

File: tools/tiara/tiara.xml [hunk 7]
```diff
@@ -0,0 +1,178 @@
...
+                <assert_contents>
+                    <has_text_matching expression=".*sequence_id*"/>
+                    <has_n_lines n="51" delta="5"/>
+                </assert_contents>
+            </output>
+        </test>
+        <test expect_num_outputs="1">
+            <param name="input" value="eukarya_fr.fasta"/>
+            <param name="taxonomy_filter" value="euk"/>
+            <param name="MIN_LEN" value="5000"/>
```
Comment (SaimMomin12) [inline tools/tiara/tiara.xml:136]:
```suggestion
            <param name="min_len" value="5000"/>
```

File: tools/tiara/tiara.xml [hunk 8]
```diff
@@ -0,0 +1,178 @@
...
+                <assert_contents>
+                    <has_text_matching expression=".*sequence_id*"/>
+                    <has_n_lines n="11" delta="5"/>
+                </assert_contents>
+            </output>
+        </test>
+        <test expect_num_outputs="1">
+            <param name="input" value="bacteria_fr.fasta"/>
+            <param name="taxonomy_filter" value="bac"/>
+            <param name="MIN_LEN" value="5000"/>
```
Comment (SaimMomin12) [inline tools/tiara/tiara.xml:149]:
```suggestion
            <param name="min_len" value="5000"/>
```

File: tools/tiara/tiara.xml [hunk 9]
```diff
@@ -0,0 +1,178 @@
...
+                </assert_contents>
+            </output>
+        </test>
+    </tests>
+    <help><![CDATA[
+What it does
+============
+
+The tiara is a Deep-learning-based approach for identification of eukaryotic sequences in the metagenomic data powered by PyTorch.
+
```
Comment (SaimMomin12) [inline tools/tiara/tiara.xml:167]: Maybe a bit more text about required inputs and generated outputs?

File: tools/tiara/macros.xml [hunk 10]
```diff
@@ -0,0 +1,26 @@
+<macros>
+    <token name="@TOOL_VERSION@">1.0.3</token>
+    <token name="@VERSION_SUFFIX@">galaxy0</token>
+
+    
+
```
Comment (SaimMomin12) [inline tools/tiara/macros.xml:6]:
```suggestion
```

**Review by SaimMomin12 — COMMENTED** [review]
Hi @Minamehr,

I've had a quick look at the PR and overall looks good. I've commented some of the changes that would be better to incorporate.

Best!

File: tools/tiara/tiara.xml [hunk 11]
```diff
@@ -27,14 +26,12 @@
         #if $min_len
             -m '$min_len'
         #end if
-
         #if $cutoff_stage1
             -p $cutoff_stage1
             #if $cutoff_stage2
             $cutoff_stage2
```
Comment (SaimMomin12) [inline tools/tiara/tiara.xml:32]: Assuming it to be nested for loop :+1:

**Review by SaimMomin12 — APPROVED** [review]

File: tools/tiara/tiara.xml [hunk 12]
```diff
@@ -0,0 +1,188 @@
...
+            <option value="bac">bacteria</option>
+            <option value="arc">archea</option>
+            <option value="euk">eukarya</option>
+            <option value="unk">unknown</option>
+            <option value="pro">prokarya</option>
+            <option value="all">all</option>
+        </param>
+        <param argument="probabilities" type="boolean" truevalue="--pr" falsevalue="" checked="false" label="Add probabilities of individual classes for each sequence."/>
+        <param argument="verbose" type="boolean" truevalue="-v" falsevalue="" checked="false" label="Display some additional messages and progress bar during classification."/>         
+        <param argument="min_len" type="integer" value="3000" optional="true" label="Minimum length of a sequence. Default: 3000 bp." help="Specify the desired minimum length in base pairs.Default value is 3000 bp and we do not recommend classifying sequences shorter than 1000 bp. "/>
```
[the remark this answers is not in the document]
Comment (bgruening) [reply tools/tiara/tiara.xml]: @Minamehr there are many more where you can add min/max ;)

**Comment by bgruening** [comment]
Thanks @Minamehr!

**Comment by bgruening** [comment]
Please do now number 3 here: https://usegalaxy-eu.github.io/posts/2020/08/22/three-steps-to-galaxify-your-tool/

**Pull request merged**